odin: vendor libraries are not available on Linux

### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$ brew config && brew doctor
HOMEBREW_VERSION: 4.3.17
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3426911432b93dc46d20d910c2f4fd33cdac62a2
Last commit: 4 days ago
Core tap JSON: 22 Aug 19:53 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: :0
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 8
SUDO_ASKPASS: /usr/libexec/openssh/gnome-ssh-askpass
Homebrew Ruby: 3.3.4 => /var/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 18.1.8
Git: 2.46.0 => /bin/git
Curl: 8.6.0 => /bin/curl
Kernel: Linux 6.10.4-201.fsync.fc40.x86_64 x86_64 GNU/Linux
OS: Fedora Linux 40.20240627.0 (Silverblue)
Host glibc: 2.39
/usr/bin/gcc: 14.2.1
/usr/bin/ruby: N/A
glibc: N/A
gcc@11: N/A
gcc: N/A
xorg: N/A
Your system is ready to brew.
```


### Verification

- [X] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

Compile a simple odin project with raylib

### What happened (include all command output)?

```
$ odin run .
/usr/bin/ld: cannot find -l:/var/home/linuxbrew/.linuxbrew/Cellar/odin/2024-08/libexec/vendor/raylib//linux/libraylib.a: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

In #164071, the prebuilt vendored libraries were removed on all platforms, but they are linked back only on macos.

--- a/Formula/o/odin.rb
+++ b/Formula/o/odin.rb
@@ -56,31 +56,46 @@ class Odin < Formula
       system "make", "unix"
     end
 
-    if OS.mac?
-      raylib_installpath = Hardware::CPU.arm? ? "vendor/raylib/macos-arm64" : "vendor/raylib/macos"
+    raylib_installpath = if OS.linux?
+      "vendor/raylib/linux"
+    elsif Hardware::CPU.intel?
+      "vendor/raylib/macos"
+    else
+      "vendor/raylib/macos-arm64"
+    end
 
-      ln_s Formula["glfw"].lib/"libglfw3.a", buildpath/"vendor/glfw/lib/darwin/libglfw3.a"
+    ln_s Formula["glfw"].lib/"libglfw3.a", buildpath/"vendor/glfw/lib/darwin/libglfw3.a"
 
-      ln_s Formula["raylib"].lib/"libraylib.a", buildpath/raylib_installpath/"libraylib.a"
-      # This is actually raylib 5.0, but upstream had not incremented this number yet when it released.
-      ln_s Formula["raylib"].lib/"libraylib.4.5.0.dylib", buildpath/raylib_installpath/"libraylib.500.dylib"
+    ln_s Formula["raylib"].lib/"libraylib.a", buildpath/raylib_installpath/"libraylib.a"
+    # This is actually raylib 5.0, but upstream had not incremented this number yet when it released.
+    ln_s Formula["raylib"].lib/shared_library("libraylib", "4.5.0"),
+      buildpath/raylib_installpath/shared_library("libraylib", "500")
 
-      resource("raygui").stage do
-        cp "src/raygui.h", "src/raygui.c"
+    resource("raygui").stage do
+      cp "src/raygui.h", "src/raygui.c"
 
-        # build static library
-        system ENV.cc, "-c", "-o", "raygui.o", "src/raygui.c",
-          "-fpic", "-DRAYGUI_IMPLEMENTATION", "-I#{Formula["raylib"].include}"
-        system "ar", "-rcs", "libraygui.a", "raygui.o"
-        cp "libraygui.a", buildpath/raylib_installpath
+      # build static library
+      system ENV.cc, "-c", "-o", "raygui.o", "src/raygui.c",
+        "-fpic", "-DRAYGUI_IMPLEMENTATION", "-I#{Formula["raylib"].include}"
+      system "ar", "-rcs", "libraygui.a", "raygui.o"
+      cp "libraygui.a", buildpath/raylib_installpath
 
-        # build shared library
-        system ENV.cc, "-o", "libraygui.dylib", "src/raygui.c",
-          "-shared", "-fpic", "-DRAYGUI_IMPLEMENTATION", "-framework", "OpenGL",
-          "-lm", "-lpthread", "-ldl",
-          "-I#{Formula["raylib"].include}", "-L#{Formula["raylib"].lib}", "-lraylib"
-        cp "libraygui.dylib", buildpath/raylib_installpath
-      end
+      # build shared library
+      args = [
+        "-o", shared_library("libraygui"),
+        "src/raygui.c",
+        "-shared",
+        "-fpic",
+        "-DRAYGUI_IMPLEMENTATION",
+        "-lm", "-lpthread", "-ldl",
+        "-I#{Formula["raylib"].include}",
+        "-L#{Formula["raylib"].lib}",
+        "-lraylib"
+      ]
+
+      args += ["-framework", "OpenGL"] if OS.mac?
+      system ENV.cc, *args
+      cp shared_library("libraygui"), buildpath/raylib_installpath
     end
 
     # By default the build runs an example program, we don't want to run it during install.
@@ -128,36 +143,37 @@ class Odin < Formula
     EOS
     system bin/"odin", "run", "miniaudio.odin", "-file"
 
+    (testpath/"raylib.odin").write <<~EOS
+      package main
+
+      import rl "vendor:raylib"
+
+      main :: proc() {
+        // raygui.
+        assert(!rl.GuiIsLocked())
+
+        // raylib.
+        num := rl.GetRandomValue(42, 1337)
+        assert(42 <= num && num <= 1337)
+      }
+    EOS
+    system bin/"odin", "run", "raylib.odin", "-file"
+
     if OS.mac?
-      (testpath/"raylib.odin").write <<~EOS
-        package main
-
-        import rl "vendor:raylib"
-
-        main :: proc() {
-          // raygui.
-          assert(!rl.GuiIsLocked())
-
-          // raylib.
-          num := rl.GetRandomValue(42, 1337)
-          assert(42 <= num && num <= 1337)
-        }
-      EOS
-      system bin/"odin", "run", "raylib.odin", "-file"
       system bin/"odin", "run", "raylib.odin", "-file",
         "-define:RAYLIB_SHARED=true", "-define:RAYGUI_SHARED=true"
-
-      (testpath/"glfw.odin").write <<~EOS
-        package main
-
-        import "core:fmt"
-        import "vendor:glfw"
-
-        main :: proc() {
-          fmt.println(glfw.GetVersion())
-        }
-      EOS
-      system bin/"odin", "run", "glfw.odin", "-file"
     end
+
+    (testpath/"glfw.odin").write <<~EOS
+      package main
+
+      import "core:fmt"
+      import "vendor:glfw"
+
+      main :: proc() {
+        fmt.println(glfw.GetVersion())
+      }
+    EOS
+    system bin/"odin", "run", "glfw.odin", "-file"
   end
 end

--- a/Formula/o/odin.rb
+++ b/Formula/o/odin.rb
@@ -9,13 +9,14 @@ class Odin < Formula
   head "https://github.com/odin-lang/Odin.git", branch: "master"
 
   bottle do
-    sha256 cellar: :any,                 arm64_sonoma:   "7651bdbe15c910bbf6804a358d5cd0fb14ccf7b232608d45379346e198f97efc"
-    sha256 cellar: :any,                 arm64_ventura:  "cde7e87c99f0c2a4ac5f4785b1786608cc783eed77e3c1cbfd05797d542cbc15"
-    sha256 cellar: :any,                 arm64_monterey: "c4a3a030528ddf0c177d33eead37b52a3942dfe2f391ccd1f6a0d1d7424dbbbf"
-    sha256 cellar: :any,                 sonoma:         "2cafdd472b3aeb6f9a463cb7a214ce89adc5d834711dc77f68fd57cc25e4df2f"
-    sha256 cellar: :any,                 ventura:        "fb26eff2b09d89cb3f222909f84c1cf6f521f7771ff13bc4e064d83bf3026d50"
-    sha256 cellar: :any,                 monterey:       "d823f59698e1d19f2984e94f70b4a16cfef51c42c1216573113c61b0a19e0445"
-    sha256 cellar: :any_skip_relocation, x86_64_linux:   "3332faaab6e3c38ca1521f7ca86087ac9d0c127dc2d49d6ce9748fb524099edc"
+    rebuild 1
+    sha256 cellar: :any,                 arm64_sonoma:   "ab56c361293b37e438e208570b0b0672bd7880fb8f34ba16dcde72e59a538263"
+    sha256 cellar: :any,                 arm64_ventura:  "0315ddfb713a361af3e9f718e0e2589af867b1fa6f5938a59bd38062f1b0001c"
+    sha256 cellar: :any,                 arm64_monterey: "3fbf54dc55468e995f7197377f4c25bd1dddbbc965976183a6685518c7e3ee38"
+    sha256 cellar: :any,                 sonoma:         "19cfebb66a3aa6270c385a1eadef48b4103bf1cf3cff240fd6d399acb86850a1"
+    sha256 cellar: :any,                 ventura:        "4850be318071e3603e3b22c014586a8b59f005d5e7c71ff5c773b2e3392041e0"
+    sha256 cellar: :any,                 monterey:       "51e6fa7c53828a3cbc95723dd38a75280fd9a524c5e92962340b692bf2d88392"
+    sha256 cellar: :any_skip_relocation, x86_64_linux:   "a007c1c815520797e87fffc4afb8ab140c228332c7b6fd2711396b9230db8850"
   end
 
   depends_on "glfw"

